compiler/rustc_codegen_cranelift/src/lib.rs

```diff
@@ -259,7 +260,7 @@ impl CodegenBackend for CraneliftCodegenBackend {
 }
 
 fn target_triple(sess: &Session) -> target_lexicon::Triple {
-    match sess.target.llvm_target.parse() {
+    match versioned_llvm_target(sess).parse() {
```

Comment:
This should probably remain unchanged. Cranelift currently doesn't consume the deployment target and cranelift-object may well expect it to be passed in separately in the future. In general Cranelift barely takes any information from the target triple. Only the cpu architecture and default calling convention for the OS currently. Even the object file format dependent way to lower TLS is passed in using a separate flag.

Reply:
Yeah, [we talked about this elsewhere](https://github.com/bytecodealliance/wasmtime/issues/8730#issuecomment-2380725262), I did it like this for now so that the commit was completely free of functional changes.

In particular, I think passing an unversioned target triple to `target-lexicon` would result in an error here, as [their current parsing code for `macosx`](https://github.com/bytecodealliance/target-lexicon/blob/7c80d459a9fdd121e9f23feb680c3db13c1baa39/src/targets.rs#L1392-L1421) assumes that the version is always present.

Reply:
I'll try to whip up a PR to fix this in `target-lexicon`, but that shouldn't block this PR IMO.

Reply:
Done, see https://github.com/bytecodealliance/target-lexicon/pull/113